src/vcbuild/warnings.h

```diff
@@ -33,6 +33,11 @@
 #pragma warning(disable:4366) // The result of the unary '&' operator may be unaligned
 #pragma warning(disable:4267) // conversion from 'size_t' to 'unsigned int', possible loss of data
 #pragma warning(disable:4310) // cast truncates constant value
+#pragma warning(disable:4838) // requires a narrowing conversion
+#pragma warning(disable:4577) // 'noexcept' used with no exception handling mode specified
+#pragma warning(disable:4805) // '|': unsafe mix of types in operation
+#pragma warning(disable:4311) // type case: pointer truncation
+#pragma warning(disable:4312) // type case: conversion from 'unsigned int' to 'void *' of greater size
```

Comment:
The latter 2 warnings should not be muted, but the unused code in os.c disabled for _WIN64.
The other warnings are not specific to the 64-bit build, so should be moved up.
When compiling the VS project (set to /W4) in VS2015, I also get a lot of these:
```
#pragma warning(disable:4456) // declaration of 'ty' hides previous local declaration
#pragma warning(disable:4458) // declaration of 'var' hides class member
#pragma warning(disable:4459) // declaration of 'var' hides global declaration
```

Reply:
> The latter 2 warnings should not be muted, but the unused code in os.c disabled for _WIN64.

https://github.com/dlang/dmd/pull/7514